pkg/referenceutil/referenceutil_test.go

```diff
@@ -290,6 +290,14 @@ func TestReferenceUtil(t *testing.T) {
 			Tag:          "latest",
 			ExplicitTag:  "",
 		},
+		"oci-archive:///tmp/build/saved-image.tar": {
+			Protocol:     "oci-archive",
+			Path:         "/tmp/build/saved-image.tar",
+			Error:        "",
+			String:       "/tmp/build/saved-image.tar",
+			Suggested:    "oci-archive-/tmp/-abcde",
```

Comment:
You likely need to do something for the suggested name of oci-archive://.

This suggested name is not valid for a container name (it is used when you do not specify a name for your container).

Maybe something in the line of:
```
func (ir *ImageReference) SuggestContainerName(suffix string) string {
	name := "untitled"
	if ir.Protocol != "" && ir.Domain == "" {
		if ir.Protocol == OCIArchive {
			name = string(ir.Protocol) + "-" + path.Base(ir.Path)[:shortIDLength]
		}else{
			name = string(ir.Protocol) + "-" + ir.String()[:shortIDLength]
		}
	} else if ir.Path != "" {
		name = path.Base(ir.Path)
	}
	return name + "-" + suffix[:5]
}

```

... although that is likely not sufficient, as filesystem might allow characters that are not safe to use as image identifiers.

Reply:
Maybe we keep it simple and only suggest ‘oci-archive-container-id’ so that we do not have to deal with local path sanitization.

Reply:
Thanks @apostasie. Let me take another look here. We load the image from the archive before and reset the image ref to the name in the image manifest before calling parse again on that reference. I shouldn't have just made this test work though. Maybe it would be better to have another type for oci-archive image references which doesn't support suggested container name as it doesn't make sense.

Reply:
I’ll have a look around as well on how this is used. 
thanks @austinvazquez !

Reply:
Should be addressed with https://github.com/containerd/nerdctl/pull/3537/commits/403499804af29636fe6168a430ee2a0ca687fc79. @apostasie, I took another look and changed how I was looking at the problem. When parsing an raw image reference which is a path for an OCI archive, we should fail and let the caller (`pkg/create`) know the reference is a path to an image archive and needs to be loaded before (re)parsing. This should resolve the quirkiness of having a image reference object which was not really a valid image reference.

Reply:
Yeah, reading through - it is a bit weird indeed, because, yep, it is not really an image ref, it is a filesystem location... - I think it is ok as you put it now.